VocaDbModel/Service/VideoServices/VideoServiceBilibili.cs

```diff
@@ -61,37 +42,37 @@ public VideoServiceBilibili()
 				return VideoUrlParseResult.CreateOk(url, PVService.Bilibili, id, VideoTitleParseResult.Empty);
 			}
 
-			var paramStr = string.Format("appkey={0}&id={1}&type=json{2}", AppConfig.BilibiliAppKey, id, AppConfig.BilibiliSecretKey);
-			var paramStrMd5 = CryptoHelper.HashString(paramStr, CryptoHelper.MD5).ToLowerInvariant();
-
-			var requestUrl = string.Format("https://api.bilibili.com/view?appkey={0}&id={1}&type=json&sign={2}", AppConfig.BilibiliAppKey, id, paramStrMd5);
+			var requestUrl = "https://api.bilibili.com/x/web-interface/view?" + (id.StartsWith("BV") ? $"bvid={id}" : $"aid={id}");
 
 			BilibiliResponse response;
 
 			try {
-				response = await JsonRequest.ReadObjectAsync<BilibiliResponse>(requestUrl, timeoutMs: 10_000, userAgent: "VocaDB/1.0 (admin@vocadb.net)");
+				response = await JsonRequest.ReadObjectAsync<BilibiliResponse>(requestUrl, timeout: TimeSpan.FromSeconds(10), userAgent: "VocaDB/1.0 (admin@vocadb.net)");
 			} catch (Exception x) when (x is HttpRequestException || x is WebException || x is JsonSerializationException || x is IOException) {
 				log.Warn(x, "Unable to load Bilibili URL {0}", url);
 				return VideoUrlParseResult.CreateError(url, VideoUrlParseResultType.LoadError, new VideoParseException(string.Format("Unable to load Bilibili URL: {0}", x.Message), x));
 			}
 
-			var authorId = response.Mid.ToString();
-            int cid = response.Cid;
+			var authorId = response.Data.Owner.Mid.ToString();
+			var aid = response.Data.Aid;
+			var bvid = response.Data.Bvid;
+			var cid = response.Data.Cid;
 
-			if (string.IsNullOrEmpty(response.Title))
+			if (string.IsNullOrEmpty(response.Data.Title))
 				return VideoUrlParseResult.CreateError(url, VideoUrlParseResultType.LoadError, "No title element");
 
-			var title = HtmlEntity.DeEntitize(response.Title);
-			var thumb = response.Pic ?? string.Empty;
-			var author = response.Author ?? string.Empty;
-			var created = response.CreatedAt;
-			var length = await GetLength(id);
+			var title = HtmlEntity.DeEntitize(response.Data.Title);
+			var thumb = response.Data.Pic ?? string.Empty;
+			var author = response.Data.Owner.Name ?? string.Empty;
+			var created = response.Data.PubDate;

```

Comment:
The publication date and time are represented in UTC, but it is displayed as local time (UTC+10:00 in my case, my clock is one hour ahead of JST, Japan Standard Time, by the way) in the song edit page. So, assuming that pubdate is 1579877678 (1/24/2020 14:54:38 in UTC), it is displayed as 1/25/2020 in my timezone, even though it is correctly stored as 1/24/2020 in the database. The same problem occurs for other PV services like SoundCloud.

Reply:
For Bilibili the case is easier, most of its producers and users live in GMT+8, so I think it would be ok to fix it to the one timezone when filling the timezone. (Anyway generally this one-day difference doesn't matter on vocadb.)

Reply:
Does the response contain timezone information? The problem is that DateTime uses local timezone, whatever that is. Timezone can be saved by using DateTimeOffset. I recommend saving the date in database as UTC. But as AgFlore said, it doesn't really matter that much.